src/controllers/auth.js

```diff
@@ -0,0 +1,51 @@
+import utils from '../utils/utils';
+import feedback from '../utils/feedback';
+import codes from '../utils/codes';
+import messages from '../utils/messages';
+
+const { passwordHasher, generateToken } = utils;
+const { successResponse, errorResponse } = feedback;
+
+/**
+   * @description Authentication class
+   */
+export default class Authentication {
+  /**
+   * @param {object} req request object
+   * @param {object} res response object
+   * @returns {object} response json object
+   * @description User sign up controller
+   */
+  static async signUp(req, res) {
+    try {
+      const {
+        firstName,
+        lastName,
+        email,
+        password,
+        gender,
+        address,
+      } = req.body;
```

Comment:
you forgot to write some attribute should be there, for instance `passportId, addressId instead of address, isVerified, rememberMe, createdAt and updatedAt`.

Reply:
I think the other attributes would be more appropriate for edit/update profile which is another story. For sign up we use the minimum required to create an account, then next step we edit/update profile.